### ISSUE TITLE
Stacks: Search + manually stack similar images

### ISSUE THREAD

**Issue by lastzero** [photoprism/photoprism#28]
Perceptual hashing is already implemented, all we need is a user interface and maybe an easy way to merge multiple files into one photo.

**Comment by rkkoszewski** [comment]
What about non-destructive merging? Where the main picture is shown as normal, but the similar pictures are still accessible from an option at the image. Just like the time-shift option in many phones, which take a burst of pictures when you make a picture and you can select the best one retrospectively while still preserving all alternatives.

**Comment by lastzero** [comment]
@rkkoszewski Yes, this is how it should look like when it's done. The database has a table for photos and files for that reason, so you can have multiple files for a photo.

**Comment by lastzero** [comment]
We've recently added a "Group by similarity" sort option to our photo search, it uses perceptual image hashes: https://demo.photoprism.org/photos?view=cards&order=similar&public=true&quality=3

**Comment by spiritbro1** [comment]
@lastzero when i try that link the photo doesnt order by similiarity just look at this photo of a girl : 
![image](https://user-images.githubusercontent.com/62529025/99207132-bd358380-27ef-11eb-82f2-b579143976af.png)
this is the first photo of the girl on the right side
![image](https://user-images.githubusercontent.com/62529025/99207162-d0e0ea00-27ef-11eb-9d73-9608db728036.png)
this is the second photo way below from the first photo can you elaborate more on what do you mean by similiarity here or is the algorithm still wrong because the order i think still not right

**Comment by timdonovanuk** [comment]
@graciousgrey am I overlooking something or is stacking not in the currently API (even though its clearly implemented)? Unstacking seems to be! I was planning to call it to stack a whole bunch of semi-duplicates I have.

**Comment by lastzero** [comment]
Stacking happens based on the file name or metadata. There is currently no way to manually stack random files without renaming / moving them or changing their metadata in any way. For example you could set a unique image ID (Exif) or document ID (XMP) and then reindex to get files stacked.

**Comment by timdonovanuk** [comment]
Ah righty, understood. Thanks @lastzero !

**Comment by GlassedSilver** [comment]
I would definitely welcome an option to not stack based on perceptional hashing, at least not automatically. Maybe with a review process and manually selecting my favored pick.

**Sample:**
![Screenshot 2022-03-28 225922](https://user-images.githubusercontent.com/1912133/160486255-e48070d0-aa64-4ef4-86ba-47c07bd8b941.png)


Three of these files certainly belong in this stack whilst `IMG_8707_edited.JPG` should be the one that gets displayed automatically.

`IMG_8708.JPG` has no business being in the stack until I give permission in my point of view. (as in: that's really just my opinion, as in my preference. I'd love to be able to set that as an option)

**Comment by graciousgrey** [comment]
@GlassedSilver PhotoPrism stacks files for the reasons mentioned here: https://docs.photoprism.app/user-guide/organize/stacks/#for-what-reasons-can-files-be-stacked.   All of them can be turned off in settings. 
Images are not yet stacked automatically due to similarity. Using edited versions as primary file is on our to do (https://github.com/photoprism/photoprism/issues/1028) :)

**Comment by GlassedSilver** [comment]
> @GlassedSilver PhotoPrism stacks files for the reasons mentioned here: https://docs.photoprism.app/user-guide/organize/stacks/#for-what-reasons-can-files-be-stacked. All of them can be turned off in settings. Images are not yet stacked automatically due to similarity. Using edited versions as primary file is on our to do (#1028) :)

I think what may have been the issue with the files I had was that they got taken shortly after one another within the same second at the same location. I turned off that setting, let's see how further imports go. :)

As for the edits becoming primaries by default... Damn I love when things I would like to address are already on the radar! :D

I've only tried this application for a few days now, but I'm already liking a lot about it. Live Photo support being one of the key features I miss in almost any other similar product.

But also that the vision in general and the wealth of features are matching my preferences. This app is starting to feel like home.

Oh, and major probs for not being subscription-only on premium features and not locking them behind some extra protected download, but being true gentlemen about it.

This is what had me even consider sponsoring this project. The more I feel pressured to do it the more likely I am to go another way, but you folks are highly respectful about this concept, so I definitely went in with a sponsorship.

I'll likely be a "repeat offender" on that front, because I think it's well worth giving back the respect that you bring towards users by not trying to be the millionth outlet to ask for a coffee a month. Makes me that much more likely to give more than that in the long run.

Sorry that I digressed. Just wanted to share my enthusiasm!

**Comment by FlorianCassayre** [comment]
I would very much welcome the possibility of manually creating/modifying stacks.

In my case, I often take similar shots to ensure that there it at least one viable photo in the pile. While I generally don't delete "duplicates" (in case I ever need them for some reason), they create noise in the photo library. I have considered archiving them, but it would make it hard to browse the remaining similar shots, or know that they even exist.

While LS hashing (or similar) would be nice, I believe that this feature should come after the aforementioned.
